test/test_nn.py

```diff
@@ -16020,41 +16020,57 @@ def test_masked_softmax(self, device):
         sizes = [(1, 1, 32), (3, 16, 310), (12, 4, 1024), (4, 2, 1200)]
         for (B, num_heads, L) in sizes:
             input = torch.randn((B, num_heads, L, L))
+            dim = input.dim() - 1
```

Comment:
you should also be checking other `dim` arguments now that you've made it an input arg

Reply:
Checking `0` and `input.dim()-1` now!